R/volcanoPlot.R

```diff
@@ -114,244 +108,381 @@ volcanoPlot <- function(contrastDF,
+        warning("symbolShape must be a vector of 3 charcter values. Assigning default values 'circle', 'circle', 'circle'.")
+        symbolShape  <- c("circle", "circle", "circle")
+
+    }
+
+    if (any(is.null(symbolColor),
+            !is.character(symbolColor),
+            length(symbolColor)  != 3,
+            length(.validate_colors(symbolColor)) != 3)) {
+        warning("symbolColor must be a vector of 3 character values. Assigning default values 'red3', 'grey25', 'deepskyblue4'.")
+        symbolColor <- c("red3", "grey25", "deepskyblue4")
+    }
+
+    if (any(is.null(transparency),
+            !is.numeric(transparency),
+            length(transparency) != 1,
+            transparency <= 0,
+            transparency > 1)) {
+        warning("transparency must be a singular value of class numeric and must be between 0 and 1. Assigning default value '0.5'.")
+        transparency <- 0.5
+    }
+
+    if (!is.null(referenceLine) &&
+        !all(is.character(referenceLine), length(referenceLine) == 1)) {
+        warning("referenceLine must be a singular value of class character or 'NULL' to disable. Assigning default value 'NULL'.")
+        referenceLine <- NULL
+    } else if (.rgbaConversion(referenceLine) == "invalid value") {
+        warning("Color specified is not valid. Assigning default value 'NULL'.")
+        referenceLine <- NULL
+    }
+
+    if (any(is.null(refLineThickness),
+            !is.numeric(refLineThickness),
+            length(refLineThickness) != 1,
+            refLineThickness < 0)) {
+        warning("refLineThickness must be a singular value of class numeric Assigning default value '2'.")
+        refLineThickness <- 2
+    }
+
+    if (any(is.null(legendPosition),
+        !is.character(legendPosition),
+        !length(legendPosition) == 1,
+        !legendPosition %in% c("top", "bottom", "left", "right", "topRight", "bottomRight", "topLeft", "bottomLeft"))) {
+        warning("legendPosition must be one value from 'top', 'bottom', 'left', 'right', 'topRight', 'bottomRight', 'topLeft', 'bottomLeft'. Assigning default value 'right'.")
+        legendPosition <- "right"
+    }
+
+    if (missing(footnote)) {
+        footnote <- NULL
+    } else if (!is.null(footnote) &&
+               !all(is.character(footnote),
+               length(footnote) == 1)) {
+        warning("footnote must be a singular value of class character or 'NULL' to disable. Assigning default value 'NULL'.")
+        footnote <- NULL
+    }
+
+    if (any(is.null(sizeByIntensity),
+            !is.logical(sizeByIntensity),
+            length(sizeByIntensity) != 1)) {
+        warning("sizeByIntensity must be a singular logical value. Assigning default value TRUE")
+        sizeByIntensity = TRUE
+    }
+
+    if (sizeByIntensity) {
+        contrastDF <- contrastDF %>%
+            dplyr::mutate("LogInt"    = dplyr::case_when(
```

Comment:
Why is this AveExpr? Shouldnt be logIntCol. You are hardcoding the variable name

Reply:
Changed.